pmda/hbond_analysis.py

```diff
@@ -0,0 +1,579 @@
+        update_selections: bool (optional)
+            Whether or not to update the acceptor, donor and hydrogen lists at
+            each frame. [True]
+
+        Examples
+        --------
+
+        The simplest use case is to allow :class:`HydrogenBondAnalysis` to
+        guess the acceptor and hydrogen atoms, and to identify donor-hydrogen
+        pairs via the bonding information in the topology::
+
+          import MDAnalysis
+          from pmda.hbond_analysis import HydrogenBondAnalysis as HBA
+
+          u = MDAnalysis.Universe(psf, trajectory)
+
+          hbonds = HBA(universe=u)
+          hbonds.run()
+
+        It is also possible to specify which hydrogens and acceptors to use
+        in the analysis. For example, to find all hydrogen bonds in water::
+
+          import MDAnalysis
+          from pmda.hbond_analysis import HydrogenBondAnalysis as HBA
+
+          u = MDAnalysis.Universe(psf, trajectory)
+
+          hbonds = HBA(universe=u,
+                       hydrogens_sel='resname TIP3 and name H1 H2',
+                       acceptors_sel='resname TIP3 and name OH2')
+          hbonds.run()
+
+        Alternatively, :attr:`hydrogens_sel` and :attr:`acceptors_sel` may be
+        generated via the :attr:`guess_hydrogens` and :attr:`guess_acceptors`.
+        This selection strings may then be modified prior to calling
+        :attr:`run`, or a subset of the universe may be used to guess the
+        atoms. For example, find hydrogens and acceptors belonging to a
+        protein::
+
+          import MDAnalysis
+          from pmda.hbond_analysis import HydrogenBondAnalysis as HBA
+
+          u = MDAnalysis.Universe(psf, trajectory)
+
+          hbonds = HBA(universe=u)
+          hbonds.hydrogens_sel = hbonds.guess_hydrogens("protein")
+          hbonds.acceptors_sel = hbonds.guess_acceptors("protein")
+          hbonds.run()
+
+        Slightly more complex selection strings are also possible.
+        For example, to find hydrogen bonds involving a protein and
+        any water molecules within 10 Å of the protein (which may be useful
+        for subsequently finding the lifetime of protein-water hydrogen bonds
+        or finding water-bridging hydrogen bond paths)::
+
+          import MDAnalysis
+          from pmda.hbond_analysis import HydrogenBondAnalysis as HBA
+
+          u = MDAnalysis.Universe(psf, trajectory)
+
+          hbonds = HBA(universe=u)
+
+          protein_hydrogens_sel = hbonds.guess_hydrogens("protein")
+          protein_acceptors_sel = hbonds.guess_acceptors("protein")
+
+          water_hydrogens_sel = "resname TIP3 and name H1 H2"
+          water_acceptors_sel = "resname TIP3 and name OH2"
+
+          hbonds.hydrogens_sel = f"({protein_hydrogens_sel}) or
+                    ({water_hydrogens_sel} and around 10 not resname TIP3})"
+          hbonds.acceptors_sel = f"({protein_acceptors_sel}) or
+                    ({water_acceptors_sel} and around 10 not resname TIP3})"
```

Comment:
@orbeckst Here is the example to set the `HydrogenBondAnalysis.donors_sel`. The current online MDAnalysis docs is still version 0.20.1, which doesn't contain the new HBond analysis module, so there's no link can be referred to.

Reply:
Ok thanks.

Btw, the devdocs are at https://www.mdanalysis.org/mdanalysis/ .